soroban-sdk/src/deploy.rs

```diff
@@ -150,10 +156,14 @@ pub struct DeployerWithAddress {
 
 impl DeployerWithAddress {
     /// Return the address of the contract defined by the deployer.
-    #[doc(hidden)]
-    /// Returns what the address of the contract will be once deployed.
+    ///
+    /// The contract doesn't have to actually be deployed in order for this to
+    /// work.
```

Comment:
Suggest taking the op to tell the story a little, and leading with when the function can be called, rather than the state of the contract.
```suggestion
    /// This function can be called at anytime, before or after the contract is deployed, because contract addresses are deterministic.
```

Reply:
Update the comment for both deployer types.